frontend/catalyst/jax_tracer.py

```
@@ -502,8 +508,125 @@ def trace_quantum_measurements(
+    such as noise present, then each tape would have different results.
+
+    Also, MidCircuitMeasure is a HybridOp, which PL does not handle at the moment.
+    Let's wait until mid-circuit measurements are better integrated into both PL
+    and Catalyst and discussed more as well."""
+    class_tracers, meas_tracers = split_tracers_and_measurements(flat_results)
+
+    # Can transforms be applied?
+    # Since transforms are a PL feature and PL does not support the same things as
+    # Catalyst, transforms may have invariants that rely on PL invariants.
+    # For example:
+    #   * mid-circuit measurements (for batch-transforms)
+    #   * that the output will be only a sequence of `MeasurementProcess`es.
+    def is_measurement(op):
+        """Only to avoid 100 character per line limit."""
+        return isinstance(op, MeasurementProcess)
+
+    is_out_measurements = map(is_measurement, meas_tracers)
+    is_all_out_measurements = all(is_out_measurements) and not class_tracers
+    is_out_measurement_sequence = is_all_out_measurements and isinstance(meas_tracers, Sequence)
+    is_out_single_measurement = is_all_out_measurements and isinstance(
+        meas_tracers, MeasurementProcess
+    )
+
+    def is_midcircuit_measurement(op):
+        """Only to avoid 100 character per line limit."""
+        # Import here to avoid circuilar dependency.
+        # pylint: disable=import-outside-toplevel
+        from catalyst.pennylane_extensions import MidCircuitMeasure
+
+        return isinstance(op, MidCircuitMeasure)
+
+    is_valid_output = is_out_measurement_sequence or is_out_single_measurement
+    is_wave_function_collapsed = any(map(is_midcircuit_measurement, tape.operations))
+    # TODO: check if the device is noisy.
+    # How?
+    is_noise_present = False
+    are_batch_transforms_valid = (
+        is_valid_output and not is_wave_function_collapsed and not is_noise_present
+    )
```

For informative transform can you raise an error? (not supported in Catalyst)

---

Raised errors here: https://github.com/PennyLaneAI/catalyst/pull/280/commits/5fe647a70d0ce814ceeeba0ac4b667f6f4f9b7c9